File: book_this-is-coding-test-with-python/part2/chap4-implementation/ex2/answer.js
function solution(n) {
  let result = 0;

  for (let i = 0; i <= n; i++) {
    if (i.toString().includes("3")) {
      result += 3600;
      continue;
    }
    for (let j = 0; j < 60; j++) {
      if (j.toString().includes("3")) {
        result += 60;
        continue;
      }
      for (let k = 0; k < 60; k++) {
        if (k.toString().includes("3")) {
          result += 1;
        }
      }
    }
  }

  return result;
}

module.exports = solution;

File: book_this-is-coding-test-with-python/part2/chap4-implementation/ex2/answer.py
n = int(input())

result = 0

for i in range(n + 1):
    if i == 3 or i == 13 or i == 23:
        result += 3600
        continue
    for j in range(60):
        if j == 3 or j == 13 or j == 23 or 39 >= j >= 30 or j == 43 or j == 53:
            result += 60
            continue
        for k in range(60):
            if k == 3 or k == 13 or k == 23 or 39 >= k >= 30 or k == 43 or k == 53:
                result += 1

print(result)
